packages/commonjs/README.md

```diff
@@ -73,53 +73,175 @@ commonjs({
 
 ### `include`
 
-Type: `String` | `Array[...String]`<br>
+Type: `string | string[]`<br>
 Default: `null`
 
 A [minimatch pattern](https://github.com/isaacs/minimatch), or array of patterns, which specifies the files in the build the plugin should operate on. By default CommonJS modules are targeted.
 
 ### `extensions`
 
-Type: `Array[...String]`<br>
+Type: `string[]`<br>
 Default: `['.js']`
 
 Search for extensions other than .js in the order specified.
 
 ### `ignoreGlobal`
 
-Type: `Boolean`<br>
+Type: `boolean`<br>
 Default: `false`
 
 If true, uses of `global` won't be dealt with by this plugin.
 
 ### `sourceMap`
 
-Type: `Boolean`<br>
+Type: `boolean`<br>
 Default: `true`
 
 If false, skips source map generation for CommonJS modules.
 
 ### `transformMixedEsModules`
 
-Type: `Boolean`<br>
+Type: `boolean`<br>
 Default: `false`
 
 Instructs the plugin whether or not to enable mixed module transformations. This is useful in scenarios with mixed ES and CommonJS modules. Set to `true` if it's known that `require` calls should be transformed, or `false` if the code contains env detection and the `require` should survive a transformation.
 
 ### `ignore`
 
-Type: `Array[...String | (String) => Boolean]`<br>
+Type: `string[] | ((id: string) => boolean)`<br>
 Default: `[]`
 
 Sometimes you have to leave require statements unconverted. Pass an array containing the IDs or an `id => boolean` function. Only use this option if you know what you're doing!
 
+### `esmExternals`
+
+Type: `boolean | string[] || ((id: string) => boolean)`
+Default: `false`
+
+Controls how imports from external dependencies are rendered. By default, all external dependencies are assumed to be CommonJS. This means they are rendered as default imports to be compatible with e.g. NodeJS where ES modules can only import a default export from a CommonJS dependency:
+
+```js
+// input
+const foo = require('foo');
+
+// output
+import foo from 'foo';
+```
+
+This is likely not desired for ES module dependencies: Here `require` should usually return the namespace to be compatible with how bundled modules are handled.
+
+If you set `esmExternals` to `true`, all external dependencies are assumed to be ES modules and will adhere to the `requireReturnsDefault` option. If that option is not set, they will be rendered as namespace imports.
+
+You can also supply an array of ids that are to be treated as ES modules, or a function that will be passed each external id to determine if it is an ES module.
+
+### `requireReturnsDefault`
+
+Type: `boolean | "auto" | "preferred" | ((id: string) => boolean | "auto" | "preferred")`<br>
+Default: `false`
+
+Controls what is returned when requiring an ES module or external dependency from a CommonJS file. By default, this plugin will render it as a namespace import, i.e.
```

Comment:
Perhaps clarify in the description directly that this is when `esmExternals` is enabled only? If I'm reading this right?